fission-core/src/capabilities/fission.rs

```diff
@@ -211,65 +214,166 @@ mod tests {
+            expiration: ucan::time::Timestamp::five_years_from_now(),
+            not_before: None
+        } // FIXME sign
+
+        pub struct AccountInfo {}
+        pub struct DnsLinkUpdate {
+            pub cid: Cid
+        }
+
+        // NOTE Just sketching an idea, don't mind me
+        // pub struct Proxy<DID: Did, T> {
+        //     pub subject: DID,
+        //     pub proxy: T,
+        //     pub prf: Vec<Cid>
+        // }
 
         // The device should now be able to use the capability, because
         // - it's got access to the account
         // - the account got delegated rights to the DNSLink
-        let invocation: Ucan = UcanBuilder::default()
-            .for_audience(server)
-            .claiming_capability(Capability::new(
-                dnslink_did.clone(),
-                FissionAbility::AccountInfo,
-                EmptyCaveat,
-            ))
-            .witnessed_by(&device_ucan, None)
-            .witnessed_by(&account_assoc_ucan, None)
-            .sign(device)?;
+        //let invocation: Invocation
+        // UcanBuilder::default()
+        //     .for_audience(server)
+        //     .claiming_capability(Capability::new(
+        //         dnslink_did.clone(),
+        //         FissionAbility::AccountInfo,
+        //         EmptyCaveat,
+        //     ))
+        //     .witnessed_by(&device_ucan, None)
+        //     .witnessed_by(&account_assoc_ucan, None)
+        //     .sign(device)?;
+
+        // 4. [dnslink -d-> account -*-> server -a-> device]
+        let account_invocation = ucan::invocation::Payload {
+            subject: account_did,
+            issuer: device,
+            audience: server,
+
+            ability: AccountInfo,
+            proofs: vec![],
+            metadata: BTreeMap::new(),
+            nonce: Nonce::generate_12(seed.as_mut()),
+            issued_at: None,
+            expiration: None,
+        }
 
-        let caps = invocation.capabilities_for(
-            &dnslink_did.clone(),
-            dnslink_did,
-            FissionAbility::AccountInfo,
-            now(),
-            &did_verifier_map,
-            &store,
-        )?;
+        let dnslink_invocation = ucan::invocation::Payload {
+            subject: dnslink_did,
+            issuer: device,
+            audience: server,
+
+            ability: DnsLinkUpdate { cid: todo!() }
+            proofs: vec![],
+            metadata: BTreeMap::new(),
+            nonce: Nonce::generate_12(seed.as_mut()),
+            issued_at: None,
+            expiration: None,
+        }
+
+        // let pbox_delegation =
+        //     Delegation::try_sign(&server_signer, varsig_header, powerbox_payload).expect("FIXME");
+
+        // let caps = invocation.capabilities_for(
+        //     &dnslink_did.clone(),
+        //     dnslink_did,
+        //     FissionAbility::AccountInfo,
+        //     now(),
+        //     &did_verifier_map,
+        //     &store,
```

Comment:
Remember how I said that I don't promise that anything _actually_ works?

I need to fix my `MemoryStore` because it depends on the `DID` having `Ord`, which they don't currently (going to use string/byte comparison).